pkg/security/security.go

```diff
@@ -0,0 +1,73 @@
+package security
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/hatchet-dev/hatchet/pkg/repository"
+
+	"github.com/rs/zerolog"
+)
+
+type SecurityCheck interface {
+	Check()
+}
+
+type DefaultSecurityCheck struct {
+	Enabled  bool
+	Endpoint string
+	Logger   *zerolog.Logger
+	Version  string
+	Repo     repository.SecurityCheckRepository
+}
+
+func NewSecurityCheck(opts *DefaultSecurityCheck, repo repository.SecurityCheckRepository) SecurityCheck {
+	return DefaultSecurityCheck{
+		Enabled:  opts.Enabled,
+		Endpoint: opts.Endpoint,
+		Logger:   opts.Logger,
+		Version:  opts.Version,
+		Repo:     repo,
+	}
+}
+
+func (a DefaultSecurityCheck) Check() {
+	if !a.Enabled {
```

Comment:
Add a `panic...recover` deferred method here so even if a random dependency panics we don't take down the engine.